assets/js/modules/analytics-4/datastore/webdatastreams.js

```diff
@@ -144,6 +174,16 @@ const baseResolvers = {
 			yield fetchGetWebDataStreamsStore.actions.fetchGetWebDataStreams( propertyID );
 		}
 	},
+	*getWebDataStreamsBatch( propertyIDs ) {
+		const registry = yield Data.commonActions.getRegistry();
+		const webdatastreams = registry.select( STORE_NAME ).getWebDataStreamsBatch( propertyIDs ) || {};
```

Comment:
If we change this to return `undefined` until all requested IDs are loaded, we can reach into the state directly for this. While we avoid doing this as much as we can, I think this is a valid case for it and given that resolvers are 1-1 with selectors that already have direct access to the state, I think this is acceptable although still not great. Otherwise, we wouldn't be able to determine what propertyIDs still need to be loaded.

Reply:
I think your approach will make it harder to understand and maintain. There is no harm in returning data that already exists in the state and request the remaining part which is not loaded yet.

Reply:
I wasn't implying that it was harmful to do so; it's simply inconsistent with how we've done things until now where state that is resolved is `undefined` until it's all there. I think this might be the first time we would potentially need multiple requests to resolve some key in state 🤔 

Generally speaking, we shouldn't be inferring the resolution status from the return value when we have dedicated selectors to check if it's been resolved so this is fine to keep.